python/ray/tune/result.py

```diff
@@ -3,6 +3,7 @@
 from __future__ import print_function
 
 from collections import namedtuple
+import yaml
```

Comment:
Make whole package depend on pyyaml just for the sake of one pretty-printing statement in ray.tune? :/

Reply:
Don't we already depend on pyyaml for parsing the configs?

Reply:
Yeah, but I think it's not explicit. To clarify, this would be a `tune/rllib` specific dependency (as with TensorFlow, opencv-python, etc as mentioned #1279)

Reply:
Btw if I may suggest: `ruamel.yaml` is generally a better option than `pyyaml` (based on YAML 1.2 and not 1.1, bans several ambiguities, able to roundtrip yaml files, also [this](http://yaml.readthedocs.io/en/latest/pyyaml.html) and [this](http://yaml.readthedocs.io/en/latest/api.html#reason-for-api-change)) -- packages that use it heavily (e.g. conda) use that instead of pyyaml. The API is the same so it's a drop-in.

Reply:
Hm, I'll leave this for now, we should be able to clean up the dependencies in #1279, or make it optional.